File: app/src/main/java/eu/toma/dev/playground/warcraft/api/IconsService.java
package eu.toma.dev.playground.warcraft.api;

import javax.inject.Inject;

import eu.toma.dev.playground.warcraft.di.Warcraft;
import io.reactivex.Observable;

@Warcraft
public class IconsService
{
    private final MediaApi api;

    @Inject
    public IconsService(MediaApi api)
    {
        this.api = api;
    }

    public Observable<?> get56Icon(){
        // TODO
        return null;
    }
}

File: app/src/main/kotlin/warcraft/api/IconsService.kt
package warcraft.api

import javax.inject.Inject

import io.reactivex.Observable
import warcraft.di.Warcraft

@Warcraft
class IconsService
@Inject
constructor(private val api: MediaApi) {

    val getIcon56 :Observable<*>?
    get()
    {
        // TODO make an api call and query the icon
        return null
    }
}
